### ISSUE TITLE
there is some package problem when I want to install gnome environment with tasksel

### ISSUE THREAD

**Issue by xlazom00** [raspberrypi/Raspberry-Pi-OS-64bit#1]
There is some package problem when I want to install gnome environment with tasksel

**Comment by solskogen** [comment]
It seems like libraspberrypi-dev conflicts with a couple of packages.

**Comment by ghollingworth** [comment]
It's probably related to us moving the /opt/vc contents into their rightful and correct places (/usr/lib for example), which means some applications will need to be rebuilt...

**Comment by PeterOGB** [comment]
See also my problem with installing openGL ES libs (which might be a dupe of this problem).

**Comment by petiepooo** [comment]
Would it make sense to have a transition package that just creates symlinks from the locations in /opt/vc to their new places in /usr/lib etc.?  It could be a dependent package for now until the other tools are updated at which point the dependency can be removed.  I would think that would help make a seamless transition while giving the other tools time to respond and be rebuilt.

**Comment by ghollingworth** [comment]
One of the reasons we have this beta version is because we know things like this need fixing.  We'd prefer for the functionality not to work on the 64 bit version until everything has been correctly fixed.

Otherwise, you end up with people relying on these links and then complaining when you finally remove them.  Or they stay there forever, which is the worst of both worlds!

A bigger pain now or continuous nagging pain forever?

**Comment by XECDesign** [comment]
> There is some package problem when I want to install gnome environment with tasksel

Uninstall pi-package first. You can use `dpkg -r` if `apt` is in a broken state already.

> It seems like libraspberrypi-dev conflicts with a couple of packages.

> See also my problem with installing openGL ES libs (which might be a dupe of this problem). It's a problem with installing the packages not an application building problem (I've not got that far yet!).

Looks like libraspberrypi-dev includes gl headers, which should be excluded on arm64. I'll take them out.

@ghollingworth, any thoughts on moving the old `include/*GL*` to `include/brcm*GL*` for armhf? It would cause some disruption.

> Would it make sense to have a transition package that just creates symlinks from the locations in /opt/vc to their new places in /usr/lib etc.? It could be a dependent package for now until the other tools are updated at which point the dependency can be removed. I would think that would help make a seamless transition while giving the other tools time to respond and be rebuilt.

I'm not sure what the transition in the armhf repo will look like yet. It's hard to move libs and dev files around without breaking things for people. But it's also required if we want multiarch to work. A package with symlinks may be one way. Another option may be to leave the existing package as is and add new one with a different name and correct paths.

**Comment by PeterOGB** [comment]
> Uninstall pi-package first. You can use `dpkg -r` if `apt` is in a broken state already.

Actually needed to unistall libraspberrypi-doc  forst then libraspberrypi-dev but then
apt --fix-broken install worked .

I can now build by openGL ES code :-)    And it works !  Thanks 
PeterO  

PS : I'll close my other report.

**Comment by XECDesign** [comment]
Your other report and OP's report are about separate issues. Op had an issue with installing gnome, which is a conflict between gnome-software and pi-package.

**Comment by PeterOGB** [comment]
Are you sure ?  Seems like the same problem with same solution.
PeterO

**Comment by solskogen** [comment]
This two packages have a conflict, so IMHO this issue should be open until
you can have both packages installed.

**Comment by XECDesign** [comment]
> Are you sure ? Seems like the same problem with same solution.

Yeah, I was able to install gnome through tasksel just by removing pi-package-data. Like you said, it doesn't fix the conflict you were seeing and that's because they're between a different set of packages.



> This two packages have a conflict, so IMHO this issue should be open until you can have both packages installed.

I think the plan is to mark pi-package and gnome-software as conflicting with each other so that installing one removes the other, or deprecating pi-package altogether. TBD...

**Comment by solskogen** [comment]
The problem is the packages libegl-dev and libraspberrypi-dev.

**Comment by PeterOGB** [comment]
> The problem is the packages libegl-dev and libraspberrypi-dev.

But as XECDesign has pointed out, that is NOT the OP's problem that this report is about.  You  have confused thing by mix it up with the libegl/libraspberrypi issue.

PeterO

**Comment by solskogen** [comment]
I stand corrected!
